witchcraft/internal/dependencyhealth/counter.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dependencyhealth
+
+import (
+	"sync"
+	"time"
+)
+
+// timeBucket measure the number of times Mark is called within the bucket interval.
+// This is necessary over using gometrics.Meter because it only updates every 5 seconds.
+// We will call Count relatively infrequently, so optimize for correctness.
```

Comment:
```suggestion
// Count is optimized for correctness since it is called relatively infrequently.
```

Can you confirm what you mean by "optimized for correctness"? I assume this is pointing to the fact that we prune the list before returning the count to have the most accurate count for all call-sites at a given time. If that is the case, I don't know if this adds much for the reader given we also prune in the `Mark` call too.

Reply:
I think I've reworded/renamed some things to make my intention clearer. I think "consistent" is a better word than "correct"